WordPress/Classes/ViewRelated/Me/MeHeaderView.m

```diff
@@ -143,14 +183,51 @@ - (UILabel *)labelForUsername
     return label;
 }
 
-- (UIImageView *)imageViewForGravatar
+- (UIImageView *)newImageViewForGravatar
 {
     CGRect gravatarFrame = CGRectMake(0.0f, 0.0f, MeHeaderViewGravatarSize, MeHeaderViewGravatarSize);
     UIImageView *imageView = [[UIImageView alloc] initWithFrame:gravatarFrame];
-    imageView.layer.cornerRadius = MeHeaderViewGravatarSize / 2.0;
+    imageView.layer.cornerRadius = MeHeaderViewGravatarSize * 0.5;
     imageView.clipsToBounds = YES;
     imageView.translatesAutoresizingMaskIntoConstraints = NO;
+    imageView.userInteractionEnabled = YES;
+    
+    UILongPressGestureRecognizer *recognizer = [[UILongPressGestureRecognizer alloc] initWithTarget:self
+                                                                                             action:@selector(handleHeaderPress:)];
+    recognizer.minimumPressDuration = MeHeaderViewMinimumPressDuration;
+    [imageView addGestureRecognizer:recognizer];
+    
     return imageView;
 }
 
+- (UIActivityIndicatorView *)newSpinner
+{
+    UIActivityIndicatorView *indicatorView = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleWhite];
+    indicatorView.hidesWhenStopped = YES;
+    indicatorView.translatesAutoresizingMaskIntoConstraints = NO;
+    
+    return indicatorView;
+}
+
+
+#pragma mark - UITapGestureRecognizer Handler
+
+- (IBAction)handleHeaderPress:(UIGestureRecognizer *)sender
+{
+    // Touch Down: Depress the gravatarImageView
+    if (sender.state == UIGestureRecognizerStateBegan) {
+        [_gravatarImageView depressAnimation];
+        return;
+    }
+    
+    // Touch Up: Normalize the gravatarImageView
+    if (sender.state == UIGestureRecognizerStateEnded) {
+        [_gravatarImageView normalizeAnimation];
+        
+        if (self.onPress) {
```

Comment:
Can we make sure that the last touch is still inside the Gravatar view bounds? I'd like to simulate "touch up inside" on this gesture recognizer so you can lift up outside of that view to cancel.